src/NuGet.Core/NuGet.Build.Tasks/GetRestoreFrameworkReferencesTask.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+
+namespace NuGet.Build.Tasks
+{
+    public class GetRestoreFrameworkReferencesTask : Task
+    {
+        /// <summary>
+        /// Full path to the msbuild project.
+        /// </summary>
+        [Required]
+        public string ProjectUniqueName { get; set; }
+
+        [Required]
+        public ITaskItem[] FrameworkReferences { get; set; }
+
+        /// <summary>
+        /// Target frameworks to apply this for. If empty this applies to all.
+        /// </summary>
+        public string TargetFrameworks { get; set; }
+
+        /// <summary>
+        /// Output items
+        /// </summary>
+        [Output]
+        public ITaskItem[] RestoreGraphItems { get; set; }
+
+        public override bool Execute()
+        {
+            var log = new MSBuildLogger(Log);
+            log.LogDebug($"(in) ProjectUniqueName '{ProjectUniqueName}'");
+            log.LogDebug($"(in) TargetFrameworks '{TargetFrameworks}'");
+            log.LogDebug($"(in) FrameworkReferences '{string.Join(";", FrameworkReferences.Select(p => p.ItemSpec))}'");
```

Comment:
What if `FrameworkReferences` is `null`?

Reply:
Can't happen, as you cannot invoke an msbuild task with a null value and this one is required.